requirements.txt

```diff
@@ -1,15 +1,14 @@
-requests
-elasticsearch>=7.0.0,<8.0.0
+Flask-Caching
+authlib
+cryptography<=3.3.2
+dacite==1.6.0
+elasticsearch==7.12.1
```

Comment:
is hard version pinning required or is the soft pinning not good enough (`elasticsearch>=7.0.0,<8.0.0`)

Reply:
hard pinning is always preferred then soft pinning or not at all. It is required for reproducible environment.